ui/client/src/views/Submissions.tsx

```diff
@@ -30,19 +32,21 @@ export const Submissions: React.FC = () => {
   const theme = useTheme();
   const addedStyle = theme.palette.mode === 'light' ? altLightModeScrollbarStyle : altDarkModeScrollbarStyle;
 
-  const { data: transactions, error, isFetching } = useQuery({
-    queryKey: ["pendingTransactions", tab, lastBlockWithTransactions],
-    queryFn: () => fetchSubmissions(tab)
+  const { data: transactions, fetchNextPage, hasNextPage, error } = useInfiniteQuery({
+    queryKey: ["submissions", tab, lastBlockWithTransactions],
+    queryFn: ({ pageParam }) => fetchSubmissions(tab, pageParam),
+    initialPageParam: undefined as IPaladinTransaction | undefined,
+    getNextPageParam: (lastPage) => { return lastPage[lastPage.length - 1] },
```

Comment:
Excellent point, it could certainly be zero. I updated the code to handle that case with the following:
```js
getNextPageParam: (lastPage) => { return lastPage.length > 0? lastPage[lastPage.length - 1] : undefined }
```